File: frontend/types/mto.ts
export interface DrawingMeta {
  drawing_no?: string;
  revision?: string;
  line_number?: string;
  nps?: string;
  material_class?: string;
  service?: string;
}

export interface MTOItem {
  item_no: number;
  category: string;
  description: string;
  size_nps?: string;
  schedule_rating?: string;
  material_spec?: string;
  end_type?: string;
  quantity: number;
  unit: string;
  length_m?: number;
  confidence?: number;
  remarks?: string;
}

export interface Summary {
  total_pipe_length_m: number;
  fittings: number;
  flanges: number;
  valves: number;
  gaskets: number;
  bolt_sets: number;
  field_welds: number;
}

export interface MTOResponse {
  drawing_meta: DrawingMeta;
  items: MTOItem[];
  summary: Summary;
  mock: boolean;
}

File: backend/app/models/mto.py
from typing import List, Optional
from pydantic import BaseModel


class DrawingMeta(BaseModel):
    drawing_no: Optional[str] = None
    revision: Optional[str] = None
    line_number: Optional[str] = None
    nps: Optional[str] = None
    material_class: Optional[str] = None
    service: Optional[str] = None


class MTOItem(BaseModel):
    item_no: int
    category: str
    description: str
    size_nps: Optional[str] = None
    schedule_rating: Optional[str] = None
    material_spec: Optional[str] = None
    end_type: Optional[str] = None
    quantity: float
    unit: str
    length_m: Optional[float] = None
    confidence: Optional[float] = None
    remarks: Optional[str] = None


class Summary(BaseModel):
    total_pipe_length_m: float = 0
    fittings: int = 0
    flanges: int = 0
    valves: int = 0
    gaskets: int = 0
    bolt_sets: int = 0
    field_welds: int = 0


class MTOResponse(BaseModel):
    drawing_meta: DrawingMeta
    items: List[MTOItem]
    summary: Summary
    mock: bool = False